pkg/expr/nodes.go

```diff
@@ -401,7 +409,44 @@ func (dn *DSNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s
 	}
 
 	var result mathexp.Results
-	responseType, result, err = s.converter.Convert(ctx, dn.datasource.Type, dataFrames, s.allowLongFrames)
+	// If the datasource node is an input to a SQL expression,
+	// the data must be in the Long format
+	if dn.isInputToSQLExpr {
+		var needsConversion bool
+		// Convert it if Multi:
+		if len(dataFrames) > 1 {
+			needsConversion = true
```

Comment:
It struck me that, at this point in the code, we know we have a Multi (and not a Wide).
And on line 425 below we know we have a Wide (not a Multi).

So instead of calling the more generic `ConvertToLong`, we could at this point call something like `ConvertMultiToLong`.
All our code could be a little more specific, and we'd need fewer code-branches.

Just thought I'd share - maybe this can be a future improvement 🤔